moped-editor/src/views/projects/projectsListView/useProjectListViewQuery/useCsvExport.js

```diff
@@ -0,0 +1,145 @@
+import React, { useState } from "react";
+import { useLazyQuery } from "@apollo/client";
+import { format } from "date-fns";
+import Papa from "papaparse";
+import {
+  Button,
+  CircularProgress,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+  Grid,
+} from "@mui/material";
+
+export const CsvDownloadDialog = ({ dialogOpen, handleDialogClose }) => (
+  <Dialog
+    open={dialogOpen}
+    onClose={handleDialogClose}
+    aria-labelledby="form-dialog-title"
+  >
+    <DialogTitle id="form-dialog-title"> </DialogTitle>
+    <DialogContent>
+      <Grid container spacing={3}>
+        <Grid item xs={2} lg={2}>
+          <CircularProgress />
+        </Grid>
+        <Grid item xs={10} lg={10}>
+          <DialogContentText>
+            Preparing download, please wait.
+          </DialogContentText>
+        </Grid>
+      </Grid>
+    </DialogContent>
+    <DialogActions>
+      <Button onClick={handleDialogClose} color="primary">
+        Cancel
+      </Button>
+    </DialogActions>
+  </Dialog>
+);
+
+/**
+ * Downloads the contents of fileContents into a file
+ * @param {string} fileContents
+ */
+const downloadFile = (fileContents, queryTableName) => {
+  const exportFileName =
+    "moped-" + queryTableName + format(Date.now(), "yyyy-MM-dd'T'HH:mm:ssxxx");
```

Comment:
@mddilley i'd like to simplify the file name a bit. currently you get something like this:
- `moped-project_list_view2023-10-10T13_22_06-04_00.csv`
and i'd like to instead have something like this:
- `moped_projects_2023-10-10.csv`

My reasoning is:
- short filenames are better
- we don't need a timestamp in the filename. if folks download multiple exports on the same day they rename the files as needed
- `project_list_view` does really mean anything to anyone except us

how does that sound to you?

Reply:
i agree with this reasoning!

Reply:
sounds great! i can't come up with a good reason to keep the timestamp around especially since the OS covers that too if it _really_ became important to someone. I'll make the change and see what everyone else thinks. 👍